signer/src/storage/in_memory.rs

```diff
@@ -153,7 +148,32 @@ impl Store {
                     .filter_map(|tx| self.raw_transactions.get(&tx.into_bytes()))
                     .filter(|sbtc_tx| sbtc_tx.tx_type == model::TransactionType::Donation)
                     .filter_map(|tx| {
-                        bitcoin::Transaction::consensus_decode(&mut tx.tx.as_slice()).ok()
+                        let txid = model::BitcoinTxId::from(tx.txid);
+                        let outputs = self.bitcoin_outputs.get(&txid)?;
+                        let prevouts = self.bitcoin_prevouts.get(&txid)?;
+                        Some(bitcoin::Transaction {
+                            version: bitcoin::transaction::Version::TWO,
+                            lock_time: bitcoin::locktime::absolute::LockTime::ZERO,
+                            input: prevouts
+                                .iter()
+                                .map(|prevout| bitcoin::TxIn {
+                                    previous_output: bitcoin::OutPoint {
+                                        txid: prevout.txid.into(),
+                                        vout: prevout.prevout_output_index,
+                                    },
+                                    script_sig: bitcoin::ScriptBuf::new(),
+                                    sequence: bitcoin::Sequence::ZERO,
+                                    witness: bitcoin::Witness::new(),
+                                })
+                                .collect(),
+                            output: outputs
+                                .iter()
+                                .map(|outpout| bitcoin::TxOut {
+                                    value: bitcoin::Amount::from_sat(outpout.amount),
+                                    script_pubkey: outpout.script_pubkey.clone().into(),
+                                })
+                                .collect(),
+                        })
```

Comment:
nit: move to separate fn and reuse it in both places?

Reply:
https://github.com/stacks-network/sbtc/pull/1559/commits/1fb10cfe6e912d828f3efacb51b9d99184a67d3a